Add check_my_repo blog post

discombobulateme opened the pull request.

Commit:
Add check_my_repo blog post

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+    - diversity and equity
+---
+
+This is a text about open source, and everything you need to be proud of your repository. Because open standards ensure not only the quality and shareability of a project with necessary documentation to make it welcoming, understandable and keeping an ongoing clear communication. Most importantly, open standards guide technology development by gently enforcing space for diversity and equity.
```
> Because open standards ensure not only the quality and shareability of a project with necessary documentation to make it welcoming, understandable and keeping an ongoing clear communication.

I don't understand this sentence

discombobulateme replied:
Does this rephrase makes more sense? 

> This is a text about open source, and everything you need to be proud of your repository. In the text I argue that by using open standards, an open source project improves its quality and shareability, since such standards exist to foster better communication between creators and consumers of the project. Most importantly, open standards guide technology development by gently enforcing space for diversity and equity.

boneskull replied:
yes

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+
+This is a text about open source, and everything you need to be proud of your repository. Because open standards ensure not only the quality and shareability of a project with necessary documentation to make it welcoming, understandable and keeping an ongoing clear communication. Most importantly, open standards guide technology development by gently enforcing space for diversity and equity.
+
+If you share my admiration for standards, optimistic that this common agreement is capable of making the world a little bit better; if you want to improve your messy project no one understands to make it more shareable; or if you just want to show off how cool you are because you have the most compliant repositories, this text is for you.
```
This is also a little difficult to read.  can you split it up into multiple sentences

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+
+## Open source as an ecosystem
+
+For some, open source is only a technical definition about code source that is open to copy, modify and distribute. Others may defend that, before you can even participate in an open source project, you must first learn to collaborate. That&#39;s because open source is not a verb nor one thing, it&#39;s an approach to software development, a workflow, a philosophy. Whatever your understanding may ...
```
```suggestion
For some, open source is only a technical definition about source code that is open to copy, modify and distribute. Others may defend that, before you can even participate in an open source project, you must first learn to collaborate. That&#39;s because open source is not a verb nor one thing, it&#39;s an approach to software development, a workflow, a philosophy. Whatever your understanding may be, it is surely more than making code publicly available.
```

you may want to link to the OSI definition of open source here

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+
+For some, open source is only a technical definition about code source that is open to copy, modify and distribute. Others may defend that, before you can even participate in an open source project, you must first learn to collaborate. That&#39;s because open source is not a verb nor one thing, it&#39;s an approach to software development, a workflow, a philosophy. Whatever your understanding may ...
+
+My personal definition is that open source is an ecosystem composed of technical things like code, but mostly of people. Just placing code on a public repository will not achieve the ideal of making it shareable, as mostly, being open means open to the &quot;other&quot; which is not me or clones of me. And this &quot;other&quot; should be able to be part of the project. But… how to include this o ...
```
I realize this is getting philosophical, but... instead of "I really don't know who might want to use or contribute to this", I think it's probably fair to say something about _audience_ here.  When you share your code, you (should) have some idea about who is going to use it.  What does this audience expect?  What are the norms in this particular community?  

The "other" lies at the edge of this audience, I think.  It's not that we've shared software for _everybody and anybody_, but rather the people who would be interested in contributing to the software should necessarily have _some_ common ground. And how you idealize the project's community will very much have an impact on the the diversity of contributors--if you focus entirely on "the code", your project may meet the OSI definition of open source, but you will likely have a homogenous contributor base.

Also (and a lot of people miss this) a project should have a well-defined scope.  Without this, you start to lose the "common ground" as mentioned above.

discombobulateme replied:
Oinnnn this is so lovely!!! Can I quote you? If so, I would add this in between both paragraphs... and this will be the 1st chaoter of my PhD o_O

> As a matter of fact, even the Open Source Initiative, one of the most active organizations since the 90's in the education and promotion of non-proprietary code, although focused on a definition particularly concerned with legal licensing, includes aspects beyond the technical, such as "No Discrimination Against Persons or Groups". 
> Christopher Hiller, recognized developer within the JavaScript community, author of two books on the language and respected maintainer of key projects like Node and Mocha, mentions the importance of including humans behind the code by design. 
> When you share your code, you (should) have some idea about who is going to use it. What does this audience expect? What are the norms in this particular community?
> The "other" lies at the edge of this audience, I think. It's not that we've shared software for everybody and anybody, but rather the people who would be interested in contributing to the software should necessarily have some common ground. And how you idealize the project's community will very much have an impact on the the diversity of contributors. If you focus entirely on "the code", your project may meet the OSI definition of open source, but you will likely have a homogenous contributor base.

boneskull replied:
haha, cool.  maybe mention me as a fellow SL employee, and replace Mocha with Appium ;)

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+
+## Mitigating communication imbalance
+
+Human communication is hard. If just a simple linear exchange requires at least a sender, a channel, a message and a receiver, what about a message that is continuously being sent in asynchronous ways, into multiple senders, receivers, channels, across the globe, through multiple languages and codes (both programming and cultural ones) as it can be an open source project?
```
```suggestion
Human communication is hard. If a simple exchange requires a sender, a channel, a message and a receiver, what about a message that is continuously being sent in asynchronous ways, into multiple senders, receivers, channels, across the globe, through multiple languages and codes (both programming and cultural ones) as it can be an open source project?
```

also please break up the second sentence

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+
+Human communication is hard. If just a simple linear exchange requires at least a sender, a channel, a message and a receiver, what about a message that is continuously being sent in asynchronous ways, into multiple senders, receivers, channels, across the globe, through multiple languages and codes (both programming and cultural ones) as it can be an open source project?
+
+Clear information should be available for everyone in the open source project ecosystem, meaning, it must have ways to achieve the message to be sent and received asynchronously through the multiple players minimizing information imbalance.
```
may be able to remove this paragraph entirely, as it's clear where you're going with it

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+
+Clear information should be available for everyone in the open source project ecosystem, meaning, it must have ways to achieve the message to be sent and received asynchronously through the multiple players minimizing information imbalance.
+
+When it comes to open source, ensuring that those outside your closest relations are on equal footing  in terms of access to information and ability to influence the project  foster collaboration principles, improve your project, and brings diversity and equity into the open source space.
```
this sentence was kind of long, feel free to use what I wrote here as an idea or w/e

```suggestion
Ensuring broad access to information and reducing information asymmetry fosters collaboration, brings diversity to a community, and ultimately improves a project.
```

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+
+When it comes to open source, ensuring that those outside your closest relations are on equal footing  in terms of access to information and ability to influence the project  foster collaboration principles, improve your project, and brings diversity and equity into the open source space.
+
+To minimize the information imbalance it is necessary to apply conscious measures in different scopes of project communication: procedurally, daily, and long-term.
```
might want to break this into subsections

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -0,0 +1,150 @@
+
+To minimize the information imbalance it is necessary to apply conscious measures in different scopes of project communication: procedurally, daily, and long-term.
+
+Procedural, as the name implies, is about the processes, the systems used. For example, maintaining a single issue tracker that is public and transparent, allowing any interested person to have open access to the issues identified by the project; having mechanisms for all participants to provide feedback or discuss possible new features; offering meet spaces that are public, scheduled in advance  ...
```
```suggestion
Procedural, as the name implies, is about the processes. One example is a public issue tracker which allows anyone to view, comment on, or create new bug reports and enhancement requests.  Another is offering scheduled and recorded public meetings (in-person or virtual), which can then be published for anyone to watch.
```

boneskull commented on content/blog/check_my_repo.md:
this seems kind of like "procedural" to me

boneskull commented on content/blog/check_my_repo.md:
```suggestion
"Long term" is the kind of information that records the history and promotes continuation of the project. Good documentation is fundamental to understand both where the project has been and where it is going.  These may include documents explaining project scope or goals, a mission, technical concepts, high-level overviews of functionality, low-level API documentation, roadmaps, learning materials, onboarding materials, project governance, team charters, etc.
```

boneskull commented on content/blog/check_my_repo.md:
> It may seem like a lot of work at first, but it will rapidly bring many benefits

Your statement does nothing to dispel that this is a lot of work.  😄   Maybe something more like "Just to be clear: this is a lot of work, but it will rapidly bring benefits..."

Also, I'm not sure where the IP and licensing mentions come from, since they weren't addressed before anywhere.

> Hopefully, it will help open source to have a diversity rate much more balanced than the 3%[^1] currently occupied by women.

I think we should be very clear (it is easy to misread this) that you cannot just use an automated tool like check-my-repo and say "well, the checks passed, so I'm good on diversity."  It takes much more work than this.  The tool is not quite a guide, but provides some guardrails on easily-identifiable problems.  But as you said yourself, OSS is about people.  We cannot code our way out of the diversity problem in OSS.

boneskull commented on content/blog/check_my_repo.md:
not sure about "and support" -- what does that mean if not part of community?

again, referencing my previous comment, we should be careful we aren't promising too much

boneskull commented on content/blog/check_my_repo.md:
> One of such tools is [check-my-repo](https://github.com/saucelabs/check-my-repo), an automated web-app that daily checks the health of all its repositories. It will help you find out whether each of your repositories contains files that will facilitate the communication and security of your projects.

can it be scheduled to run at different intervals?  maybe needn't mention "daily"

> It will help you find out whether each of your repositories contains files that will facilitate the communication and security of your projects.

Does it do this, or the opposite?

boneskull commented on content/blog/check_my_repo.md:
> It is an automated web application built upon [Repolinter](https://todogroup.github.io/repolinter/) that verifies the main necessary parameters to comply with open source best practices: from Readme to License. Automating this verification helps with maintenance, forwarding compliant software projectseasier to service, test, upgrade, maintain and contribute.

- maybe replace "comply" with "adhere"; nobody is issuing commands
- "open source best practices" might want to be "best practices for building inclusive OSS communities" or something, since it's kind of vague what the former means
- The last sentence doesn't quite make sense and has a typo.    maybe replace with "intended to be part of a CI workflow to avoid mistakes and regressions" or something

boneskull replied:
looks like the next sentence says as much; maybe this one can be removed, or changed to say something generic like "where feasible, automation reduces maintenance burden"

boneskull commented on content/blog/check_my_repo.md:
- you will need a github account.  
- screenshots will be helpful for those new to github.

boneskull commented on content/blog/check_my_repo.md:
are we sure they aren't already enabled?

boneskull commented on content/blog/check_my_repo.md:
is gh-pages automatically configured in a fork?

boneskull commented on content/blog/check_my_repo.md:
use backticks around the cron job def

boneskull commented on content/blog/check_my_repo.md:
hmm... binary files like images are common.  executables are not

boneskull commented on content/blog/check_my_repo.md:
- "use several components" may want to be "include dependencies on other OSS projects"

otherwise, it's unclear who is to "combine" licenses? the maintainer? the user?  

you might want to provide some offsite link to an overview of licenses, and what they mean. 

it's unclear what the rule actually does here

boneskull commented on content/blog/check_my_repo.md:
changelog and semver are not mutually exclusive.  many projects have both.  changelogs are a list of what exactly changed, and semver is just the version of the project.  may want to link to semver, or just omit semver from this paragraph.  

other reasons for the changelog include documenting breaking changes (and migration paths) and security fixes

boneskull commented on content/blog/check_my_repo.md:
this should be broken up.  maybe instead of "considering cognitive aspects" you could just suggest a CONTRIBUTING.md as a baseline and encourage people to create videos or other walkthroughs

boneskull commented on content/blog/check_my_repo.md:
what languages does this tool support?  we should mention that

boneskull commented on content/blog/check_my_repo.md:
I'm not sure we can say with confidence that tests have a measurable impact on project security other than fuzz testing

boneskull commented on content/blog/check_my_repo.md:
instead of "some of whom you may never even meet" maybe be more straightforward like _you risk alienating potential future contributors_

boneskull commented on content/blog/check_my_repo.md:
re filenames, usually these are wrapped in backticks

boneskull commented on content/blog/check_my_repo.md:
heh, okay, but maybe again we need to be clear that check-my-repo will only show you if there's some obvious deficiency.  to get to that beautiful repo, you need to put in some work!

boneskull commented on content/blog/check_my_repo.md:
regarding the points you made earlier around open, transparent lines of communication, I think it'd be great to be able (where it makes sense) to reference those points in the list of rules above.

discombobulateme replied:
hummmm I do not check for them though... thinking how to add it...

boneskull commented on content/blog/check_my_repo.md:
again, you may want "adhere" or some other word than "comply"

boneskull commented on content/blog/check_my_repo.md:
it shouldn't!

boneskull commented on content/blog/check_my_repo.md:
do we have any links?

discombobulateme replied:
Yes, in the footnotes.

boneskull commented on content/blog/check_my_repo.md:
analyzing.

also use markdown for links and consider using archive.org.  should probably link to CHAOSS, LF, etc.

rather strange that the linter didn't automatically fix the issues.  I swear I set something up.  how did you commit this changeset?

discombobulateme replied:
I am lost in this o_O is [1]: not a markdown for footnote? 
What do you mean by that? "consider using archive.org. should probably link to CHAOSS, LF, etc." it is a LF research. Lost lol.
I committed via git...

boneskull replied:
- I mean instead of bare links (`https://...`) use markdown `[text](https://...)`
- may want to change the wired article link and LF foundation link to point to archive.org copy. up to you though
- link to CHAOSS and LF means exactly that. `[CHAOSS](http://chaoss.url)`

boneskull reviewed (commented):
this is good!! please take into consideration my suggestions/discussion

Commit:
Change having to making code

Co-authored-by: Christopher Hiller <boneskull@boneskull.com>

Commit:
Change having to making code

Co-authored-by: Christopher Hiller <boneskull@boneskull.com>

Commit:
Improved text by cutting a few words

Co-authored-by: Christopher Hiller <boneskull@boneskull.com>

Commit:
Improved text by cutting a few words

Co-authored-by: Christopher Hiller <boneskull@boneskull.com>

Commit:
Improved text based on Chris discussion (comments)

Commit:
Improved text based on Chris discussion (comments)

boneskull commented:
@discombobulateme Many linting failures here... in your working copy, run `npm install` then `npm run format`.  This should fix all the problems.  You can then commit the changed files and push.

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -13,14 +13,23 @@ tags:
 
 For some, open source is only a technical definition about source code that is open to copy, modify and distribute. Others may defend that, before you can even participate in an open source project, you must first learn to collaborate. That&#39;s because open source is not a verb nor one thing, it&#39;s an approach to software development, a workflow, a philosophy. Whatever your understanding may ...
 
+As a matter of fact, even the [Open Source Initiative](https://opensource.org/osd), one of the most active organizations since the 90's in the education and promotion of non-proprietary code, although focused on a definition particularly concerned with legal licensing, includes aspects beyond the technical, such as "No Discrimination Against Persons or Groups". 
```
Maybe "free and open source software" instead of "non-proprietary code"?

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -13,14 +13,23 @@ tags:
 
+As a matter of fact, even the [Open Source Initiative](https://opensource.org/osd), one of the most active organizations since the 90's in the education and promotion of non-proprietary code, although focused on a definition particularly concerned with legal licensing, includes aspects beyond the technical, such as "No Discrimination Against Persons or Groups". 
+
+Christopher Hiller, recognized developer within the JavaScript community, author of [two books](https://www.goodreads.com/author/show/7135237.Christopher_Hiller) on the language and respected maintainer of key projects like Node and Mocha, mentions the importance of including humans behind the code by design. 
```
ah, uh, those books are old, maybe just remove the links or mention of it 😄

boneskull commented on content/blog/check_my_repo.md:
```diff
@@ -13,14 +13,23 @@ tags:
+Christopher Hiller, recognized developer within the JavaScript community, author of [two books](https://www.goodreads.com/author/show/7135237.Christopher_Hiller) on the language and respected maintainer of key projects like Node and Mocha, mentions the importance of including humans behind the code by design. 
+
+>When you share your code, you (should) have some idea about who is going to use it. What does this audience expect? What are the norms in this particular community?
+>The "other" lies at the edge of this audience, I think. It's not that we've shared software for everybody and anybody, but rather the people who would be interested in >contributing to the software should necessarily have some common ground. And how you idealize the project's community will very much have an impact on the the diversity of contributors. If you focus entirely on "the code", your p ...
```
formatting / quoting is weird here

boneskull reviewed (commented):
are you finished with this?

discombobulateme commented:
> npm run format

humpf... format command pointed out 2 minor extra whitespaces, then solved. It keeps showing me several mistakes on other blog post: ubuntu_autoinstall_and_iac_data_centers_operations.md
Weird...

Commit:
Delete extra space

Commit:
Merge branch 'check-my-repo-blog' of github.com:saucelabs/saucelabs.github.io into check-my-repo-blog

Commit:
A few wording change and delete books links

Commit:
A few wording change and delete books links

boneskull commented:
@diemol ehh, linting is busted.  I'll look into it. but lint failures should not prevent merging

boneskull committed:
chore: fix formatting for yaml files (#95)

- corrected un-parsable yaml in `ubuntu_autoinstall_and_iac_data_centers_operations.md`; reformatted
- changed `eslint-plugin-yml` ruleset to use `yml/prettier` which should prevent conflicts
- removed conflicting rule about indents
- removed conflicting setting in `.editorconfig` about trailing whitespace in markdown files

diemol committed:
Merge branch 'dev' into check-my-repo-blog

diemol committed:
Update linter.yml (#96)

* Update linter.yml

* Update linter.yml

* Linting file

* Adding package-lock.json

diemol committed:
Merge branch 'dev' into check-my-repo-blog

diemol committed:
Updating preview image

diemol committed:
Merge branch 'dev' into check-my-repo-blog

boneskull commented:
@discombobulateme can you rebase this onto `dev` then force-push yr branch?

Commit:
Change links, add markdown link to footnotes

Commit:
Merge branch 'check-my-repo-blog' of github.com:saucelabs/saucelabs.github.io into check-my-repo-blog
